angel_system/impls/detect_activities/swinb/swinb_detect_activities.py

```diff
@@ -99,8 +99,16 @@ def get_model(self) -> torch.nn.Module:
             model = model.eval()
 
             # Transfer the model to the requested device
-            model_device = torch.device(self._torch_device)
-            model = model.to(device=model_device)
+            if self._torch_device != 'cpu':
+                if torch.cuda.is_available():
+                    model_device = torch.device(device=self._torch_device)
+                    model = model.to(device=model_device)
+                else:
+                    raise RuntimeError(
+                        "Use of CUDA requested but not available."
+                    )
+            else:
+                model_device = torch.device(self._torch_device)
```

Comment:
I had mentioned to Sri that this more complicated block shouldn't be necessary. If we try to send the model to an invalid device, we should get an appropriate error. There are also other devices than just CPU and CUDA that I don't see why we specifically need to cut out of the picture (not that we're likely to use them). Was this change based on an experience you had?